pkg/topology/topology.go

```diff
@@ -0,0 +1,116 @@
+package topology
+
+import (
+	specs "github.com/servicemeshinterface/smi-sdk-go/pkg/apis/specs/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NameNamespace is a key for referencing unique resources.
+type NameNamespace struct {
```

Comment:
Shouldn't it be called a `topology.Key` then? I'm not fond of having a key called by what it contains 🤔

Reply:
Also this would make it more resilient to change (unlikely to happen but hey) :)